alpha/apps/kaltura/modules/extwidget/actions/playManifestAction.class.php

```diff
@@ -1323,4 +1357,23 @@ protected function setPlaylistFlavorAssets($durations, $mediaEntryId)
 		$this->deliveryAttributes->setStorageId(null);
 		$this->deliveryAttributes->setFlavorAssets($filteredFlavorAssets);
 	}
+
+	/**
+	 * @param $a
+	 * @param $b
+	 * @return int
+	 */
+	public static function sortFlavorsByFrameSizeAndBitrate($a, $b)
+	{
+		/* @var $a flavorAsset */
+		/* @var $b flavorAsset */
+		if ($a->getHeight() * $a->getWidth() == ($b->getHeight() * $b->getWidth()))
+		{
+			return $a->getBitrate() - $b->getBitrate();
```

Comment:
In case the values are the same make sure to prioritize based on their original order in the array otherwise, you will get an inconsistent order for different requests.

SAme applies to "return ($a->getHeight() * $a->getWidth()) - ($b->getHeight() * $b->getWidth());"

Reply:
Original order doesn't matter if the values are the same since in the end of the filtering you will be left with only one flavor to play.

Reply:
fixed it anyway